snapcraft/internal/repo/snaps.py

```diff
@@ -94,6 +94,77 @@ def in_store(self):
                 self._is_in_store = False
         return self._is_in_store
 
+    def _get_installed_path(self):
+        return os.path.join("/snap", self.name, "current")
+
+    def _get_installed_meta_snap_yaml(self):
+        if not self.installed:
+            return None
+
+        yaml_path = os.path.join(self._get_installed_path(), "meta", "snap.yaml")
+        if not os.path.exists(yaml_path):
+            return None
+
+        return yaml_utils.load(open(yaml_path, "r"))
+
+    def _translate_installed_path(self, path):
+        if path.startswith("$SNAP"):
+            return path.replace("$SNAP", self._get_installed_path())
+        elif path.startswith("/"):
+            return os.path.join(self._get_installed_path(), path.lstrip("/"))
+
+        logger.warn("warning: unexpected slot path found ({})".format(path))
+        return path
+
+    def _translate_installed_dirs(self, paths: Set[str]):
+        translated = set()
+
+        for path in paths:
+            path = self._translate_installed_path(path)
+            translated.add(path)
+
+        return translated
+
+    def get_slot_content_dirs(self, slot_name: str, installed_path=True) -> Set[str]:
```

Comment:
This feels a bit too specific for this class, maybe this can be get_slots and returns a collection class of slot objects?
Maybe too much work, but let's think about it

Reply:
I agree.  I was thinking to formalize some classes to handle the marshal/unmarshal meta/snap.yaml.  In the case of the snap the could just request the "SnapMeta" or whatever, then the user can sort out what they want from it.  Could be shared with the various users of snap.yaml (_packer, snapcache, etc.).